docs/antrea-ipam.md

```diff
@@ -262,6 +289,36 @@ router should provide the network connectivity for these VLANs. Only a single IP
 be included in the Namespace annotation. In the future, annotation of up to two pools for
 IPv4 and IPv6 respectively will be supported.
 
+### Flexible IPAM design
+
+When the `AntreaIPAM` feature gate is enabled, `antrea-controller` will watch IPPool CRs and
+StatefulSets from kube-apiserver, and `antrea-agent` will register the IPAM driver to CNI server.
+
+#### On IPPool CR create/update event
+
+`antrea-controller` will update IPPool counters, and periodically clean up stale IP addresses.
+
+#### On StatefulSet create event
+
+`antrea-controller` will check the Antrea IPAM annotations on the StatefullSet, and preallocate
+IPs from the specified IPPool for the StatefullSet Pods
+
+#### On StatefulSet delete event
+
+`antrea-controller` will clean up IP allocations for this StatefulSet.
+
+#### On Pod create
+
+`antrea-agent` will receive a CNI add request, and it will check the Antrea IPAM annotations and
+allocate an IP for the Pod, which can be a pre-allocated IP StatefulSet IP, a user-specified IP,
+or the next available IP in the specified IPPool.
+
+#### On Pod delete
+
+`antrea-agent` will receive a CNI del request and release the IP allocation from the IPPool.
+If the IP is a pre-allocated StatefulSet IP, it will remain pre-allocated status thus the Pod will
```

Comment:
"it will stay in the `pre-allocated` status"

Reply:
Fixed. Thanks.